SofaKernel/modules/SofaBaseMechanics/MechanicalObject.inl

```diff
@@ -1087,12 +1085,15 @@ void MechanicalObject<DataTypes>::init()
         numa_set_preferred(this->getContext()->getProcessor()/2);
 #endif
 
-    //Look at a topology associated to this instance of MechanicalObject by a tag
-    this->getContext()->get(m_topology, this->getTags());
-    // If no topology found, no association, then look to the nearest one
-    if(m_topology == NULL)
-        m_topology = this->getContext()->getMeshTopology();
+    if (!l_topology && d_useTopology.getValue())
+    {
+        l_topology.set( this->getContext()->getActiveMeshTopology() );
+    }
 
+    if (l_topology)
+    {
+        sout << "Initialization with topology " << l_topology->getTypeName() << " " << l_topology->getName() << " ( " << l_topology->getNbPoints() << " points)" << sendl; // << (l_topology->hasPos() ? "WITH" : "WITHOUT") << " positions)"
```

Comment:
The purpose was the have a developer-friendly message at the top of the component's log that would catch his attention if the implicitly-defined-with-tricky-rules association to a topology was incorrect. But it could indeed be removed.
We did not switch to the new msg_ formating due to conflicts with earlier changes on our branch and pending review of this quite complex new logger mechanism. But feel free to patch it if you know the right way to make it stay only visible if you look for it within the component's log and not the general console.